### PR TITLE
feat: onboard ga4 hybrid mode

### PR DESCRIPTION
**Pull request by mihir-4116**
## PR Description

Please include a summary of the change along with the relevant motivation and context.

## Notion ticket

Ticket link

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/786)

### CODE REVIEW

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: src/core/analytics.js [hunk 1]
```diff
@@ -13,6 +13,7 @@
 /* eslint-disable no-param-reassign */
 import Emitter from 'component-emitter';
 import { parse } from 'component-querystring';
+import merge from 'lodash.merge';
```
[the remark this answers is not in the document]
Comment (mihir-4116) [reply src/core/analytics.js]: updated

**Review by MoumitaM — APPROVED** [review]

File: src/core/analytics.js [hunk 2]
```diff
@@ -742,6 +766,24 @@ class Analytics {
       // convert integrations object to server identified names, kind of hack now!
       transformToServerNames(rudderElement.message.integrations);
```
Comment (krishna2020) [inline src/core/analytics.js:768]: Add same comment of example objects & expected merge object

Comment (mihir-4116) [reply src/core/analytics.js:768]: [ref](https://github.com/rudderlabs/rudder-sdk-js/pull/805#discussion_r1055137246)

**Review by krishna2020 — APPROVED** [review]

**Review by MoumitaM — DISMISSED** [review]

**Comment by bardisg** [comment]
@mihir-4116 after discussion with @krishna2020 I will pick up the changes for my review comments as we want to use this as an example for unit testing. Will arrange a session next week to go through te changes I will make tomorrow

File: .size-limit.js [hunk 3]
```diff
@@ -7,13 +7,13 @@ module.exports = [
...
     gzip: true,
-    limit: '37 kB',
+    limit: '40 kB',
   },
   {
     name: 'Core - CDN (modern only)',
     path: 'dist/modern/rudder-analytics.min.js',
     gzip: true,
-    limit: '35 kB',
+    limit: '40 kB',
```
[the remark this answers is not in the document]
Comment (bardisg) [reply .size-limit.js]: done

File: .size-limit.js [hunk 4]
```diff
@@ -7,13 +7,13 @@ module.exports = [
     name: 'Core - CDN',
     path: 'dist/legacy/rudder-analytics.min.js',
     gzip: true,
-    limit: '37 kB',
+    limit: '40 kB',
```
[the remark this answers is not in the document]
Comment (bardisg) [reply .size-limit.js]: done

File: src/core/analytics.js [hunk 5]
```diff
@@ -742,6 +766,49 @@ class Analytics {
...
+      After Merge
+      rudderElement.message.integrations = {
+         "Google Analytics 4": {
+            "sessionId": "1669961395"
+        },
+        "AM": false
+      }
+      */
+
+      const tempIntegrationsData = R.clone(this.integrationsData);
```
[the remark this answers is not in the document]
Comment (bardisg) [reply src/core/analytics.js]: done

[the remark this answers is not in the document]
Comment (bardisg) [reply src/core/analytics.js]: done

File: src/core/analytics.js [hunk 6]
```diff
@@ -742,6 +766,49 @@ class Analytics {
       // convert integrations object to server identified names, kind of hack now!
       transformToServerNames(rudderElement.message.integrations);
 
+       /*
+      Example :
```
[the remark this answers is not in the document]
Comment (bardisg) [reply src/core/analytics.js]: done

File: src/core/analytics.js [hunk 7]
```diff
@@ -365,6 +388,7 @@ class Analytics {
     if (object.clientIntegrationObjects.every((intg) => !intg.isReady || intg.isReady())) {
       // Integrations are ready
       // set clientIntegrationsReady to be true
+      object.prepareDataForIntegrationsObj(object.clientIntegrationObjects);
```
[the remark this answers is not in the document]
Comment (bardisg) [reply src/core/analytics.js]: done

File: src/core/analytics.js [hunk 8]
```diff
@@ -341,6 +342,28 @@ class Analytics {
     }
   }
 
+  /**
+   * Prepares the data for integrationsObj
+   *
+   * @param {*} integration
+   * @param {*} integrationInstance
```
[the remark this answers is not in the document]
Comment (bardisg) [reply src/core/analytics.js]: done

File: src/core/analytics.js [hunk 9]
```diff
@@ -341,6 +342,28 @@ class Analytics {
...
+   * @memberof Analytics
+   */
+  prepareDataForIntegrationsObj(integrationInstances) {
```
[the remark this answers is not in the document]
Comment (bardisg) [reply src/core/analytics.js]: done

**Review by bardisg — APPROVED** [review]

**Review by MoumitaM — DISMISSED** [review]

**Review by saikumarrs — DISMISSED** [review]

**Comment by bardisg** [comment]
@mihir-4116 do we still have anything outstanding on this one?

**Comment by mihir-4116** [comment]
> @mihir-4116 do we still have anything outstanding on this one?

we can consider this as final one

**Review by bardisg — APPROVED** [review]

**Review by MoumitaM — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

File: .size-limit.js [hunk 10]
```diff
@@ -7,7 +7,7 @@ module.exports = [
     name: 'Core - CDN',
     path: 'dist/legacy/rudder-analytics.min.js',
     gzip: true,
-    limit: '36.5 kB',
+    limit: '38 kB',
```
Comment (saikumarrs) [inline .size-limit.js:10]: Why did we increase the size again?
CC @bardisg

**Pull request merged**